File: FrontEnd/src/app/features/reservations/pickupPoints/classes/dtos/pickupPoint-read-dto.ts
import { CoachRouteAutoCompleteVM } from '../../../coachRoutes/classes/view-models/coachRoute-autocomplete-vm'
import { Metadata } from 'src/app/shared/classes/metadata'
import { PortAutoCompleteVM } from '../../../ports/classes/view-models/port-autocomplete-vm'
import { SimpleEntity } from 'src/app/shared/classes/simple-entity'

export interface PickupPointReadDto extends Metadata {

    // PK
    id: number
    // Object fields
    coachRoute: CoachRouteAutoCompleteVM
    destination: SimpleEntity
    port: PortAutoCompleteVM
    // Fields
    description: string
    exactPoint: string
    time: string
    remarks: string
    isActive: boolean

}

File: API/Features/Reservations/PickupPoints/Dtos/PickupPointReadDto.cs
using API.Features.Reservations.CoachRoutes;
using API.Infrastructure.Classes;
using API.Infrastructure.Interfaces;

namespace API.Features.Reservations.PickupPoints {

    public class PickupPointReadDto : IBaseEntity, IMetadata {

        // PK
        public int Id { get; set; }
        // FKs
        public int CoachRouteId { get; set; }
        public int DestinationId { get; set; }
        public int PortId { get; set; }
        // Fields
        public string Description { get; set; }
        public string ExactPoint { get; set; }
        public string Time { get; set; }
        public string Remarks { get; set; }
        public bool IsActive { get; set; }
        // Metadata
        public string PostAt { get; set; }
        public string PostUser { get; set; }
        public string PutAt { get; set; }
        public string PutUser { get; set; }
        // Navigation
        public CoachRouteAutoCompleteVM CoachRoute { get; set; }
        public SimpleEntity Destination { get; set; }
        public SimpleEntity Port { get; set; }

    }

}